ledfx/effects/hierarchy.py

```diff
@@ -0,0 +1,114 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class Filter(AudioReactiveEffect, GradientEffect):
+    NAME = "Hierarchy"
+    CATEGORY = "Simple"
+    HIDDEN_KEYS = [
+        "background_color",
+        "background_brightness",
+        "blur",
+        "mirror",
+        "flip",
+        "gradient_roll",
+        "gradient",
+        "frequency_range",
+        "brightness",
+    ]
+    ADVANCED_KEYS = []
+
+    CONFIG_SCHEMA = vol.Schema(
+        {
+            vol.Optional(
+                "color_lows",
+                description="Color of low, bassy sounds",
+                default="#FF0000",
+            ): validate_color,
+            vol.Optional(
+                "color_mids",
+                description="Color of midrange sounds",
+                default="#00FF00",
+            ): validate_color,
+            vol.Optional(
+                "color_high",
+                description="Color of high sounds",
+                default="#0000FF",
+            ): validate_color,
+            vol.Optional(
+                "brightness_boost",
+                description="Multiplier for the Brightness",
+                default=1.0,
+            ): vol.All(vol.Coerce(float), vol.Range(min=0.0, max=4.0)),
+            vol.Optional(
+                "switch_threshold_lows",
+                description="If Lows are below this value, Mids are used.",
+                default=0.05,
+            ): vol.All(vol.Coerce(float), vol.Range(min=0.0, max=1.0)),
+            vol.Optional(
+                "switch_threshold_mids",
+                description="If Mids are below this value, Highs are used",
+                default=0.05,
+            ): vol.All(vol.Coerce(float), vol.Range(min=0.0, max=1.0)),
+            vol.Optional(
+                "switch_time",
+                description="Time Lows/Mids have to be below threshold before switch",
+                default=0.1,
+            ): vol.All(vol.Coerce(float), vol.Range(min=0.0, max=1.0)),
+        }
+    )
+
+    def on_activate(self, pixel_count):
+        self.filtered_power = 0
+        self.last_low = 0
+        self.last_mid = 0
+        self.color = np.array(parse_color("#000000"))
+
+    def config_updated(self, config):
+        self.power_func = "Lows (beat+bass)"
+        self.color_low = np.array(parse_color(self._config["color_lows"]))
+        self.color_mids = np.array(parse_color(self._config["color_mids"]))
+        self.color_high = np.array(parse_color(self._config["color_high"]))
+
+    def audio_data_updated(self, data):
+        # use Lows (beat+bass)
+        self.power_func = self.POWER_FUNCS_MAPPING["Lows (beat+bass)"]
+        self.filtered_power = getattr(data, self.power_func)()
+        if self.filtered_power > self._config["switch_threshold_lows"]:
+            self.color = self.color_low
+            self.last_low = timeit.default_timer()
+
+        elif (
+            timeit.default_timer() - self.last_low
```

Comment:
I would also recommend a single call to the timeit function at the top of the method, if you are not measuring internal behaviour.